uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -0,0 +1,516 @@
+              $ctrl.leaveRequest.to_date_type = $ctrl.leaveRequest.from_date_type;
+            }
+
+            if (!$ctrl.uiOptions.showBalance) {
+              return
+            }
+
+            $ctrl.loading.calculateBalanceChange = true;
+            if ($ctrl.leaveRequest.from_date && $ctrl.leaveRequest.to_date) {
+              $ctrl.calculateBalanceChange().then(function () {
+                $ctrl.loading.dateChangesBalance = false;
+              });
+            }
+          });
+      };
+
+      /**
+       * Calculate change in balance, it updates local balance variables
+       *
+       * @return {Promise}
+       */
+      $ctrl.calculateBalanceChange = function () {
+        if ($ctrl.uiOptions.selectedToType) {
+          $ctrl.leaveRequest.to_date_type = $ctrl.uiOptions.selectedToType.name;
+        }
+
+        if ($ctrl.uiOptions.selectedFromType) {
+          $ctrl.leaveRequest.from_date_type = $ctrl.uiOptions.selectedFromType.name;
+
+          if (!$ctrl.uiOptions.multipleDays) {
+            $ctrl.leaveRequest.to_date_type = $ctrl.leaveRequest.from_date_type;
+          }
+        }
+
+        $ctrl.leaveRequest.from_date = convertDateFormatToServer($ctrl.uiOptions.fromDate);
+        $ctrl.leaveRequest.to_date = convertDateFormatToServer($ctrl.uiOptions.toDate);
+
+        var params = _.pick($ctrl.leaveRequest, ['contact_id', 'from_date', 'from_date_type', 'to_date', 'to_date_type']);
+
+        //todo to remove in future when this call is consistent with leaverequest db fields name
+        params = _.mapKeys(params, function (value, key) {
+          if (key == 'from_date_type') {
+            return 'from_type';
+          } else if (key == 'to_date_type') {
+            return 'to_type';
+          }
+
+          return key;
+        });
+
+        $ctrl.error = undefined;
+        return LeaveRequest.calculateBalanceChange(params)
+          .then(function (balanceChange) {
+            if (balanceChange) {
+              $ctrl.balance.change = balanceChange;
+              //the change is negative so adding it will actually subtract it
+              $ctrl.balance.closing = $ctrl.balance.opening + $ctrl.balance.change.amount;
+              rePaginate();
+            } else {
+              console.log('$ctrl.calculateBalanceChange', params, balanceChange);
+            }
+          })
+          .catch(function (errors) {
+            if (errors.error_message)
+              $ctrl.error = errors.error_message;
+            else {
+              $ctrl.error = errors;
+            }
+          });
+      };
+
+      /**
+       * This method will be used on the view to return a list of available
+       * leave request day types (1/2 PM, Non working day, etc) for the given date
+       * (which is the date selected by the user via datepicker)
+       *
+       * If no date is passed, then no list is returned
+       *
+       * @param  {Date} dateParam
```

Comment:
You are missing the `@param` for `isFrom`

Also, why not calling the first param just `date`? We know it's a param, what's the point of calling it `date*Param*`